DllImportGenerator/DllImportGenerator/Marshalling/MarshallingGenerator.cs

```diff
@@ -250,7 +250,7 @@ private static IMarshallingGenerator ValidateByValueMarshalKind(StubCodeContext
                 // Must go before the cases that do not explicitly check for marshalling info to support
                 // the user overridding the default marshalling rules with a MarshalUsing attribute.
                 case { MarshallingAttributeInfo: NativeMarshallingAttributeInfo marshalInfo }:
-                    return CreateCustomNativeTypeMarshaller(info, context, marshalInfo);
+                    return CreateCustomNativeTypeMarshaller(info, context, marshalInfo, options);
```

Comment:
We could pair it, but it would feel a little odd. I think a better model would be to use the design in #1119 where the method with this giant switch is in a non-static class, and the options are (or could be) passed into the object's constructor instead of passed down as a parameter.

Reply:
Okay. Let's focus on the extensibility PR then to address this.